include/dlaf/eigensolver/backtransformation/impl-t2b.h

```diff
@@ -0,0 +1,317 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2021, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <type_traits>
+
+#include <hpx/include/util.hpp>
+
+#include "dlaf/blas/tile.h"
+#include "dlaf/common/range2d.h"
+#include "dlaf/common/round_robin.h"
+#include "dlaf/eigensolver/backtransformation/api.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/matrix/panel.h"
+#include "dlaf/matrix/tile.h"
+#include "dlaf/sender/traits.h"
+#include "dlaf/types.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf::eigensolver {
+namespace internal {
+
+template <class T>
+static constexpr bool is_complex_v = std::is_same_v<T, ComplexType<T>>;
+
+template <class T>
+static SizeType nrSweeps(const SizeType m) {
+  return std::max<SizeType>(0, is_complex_v<T> ? m - 1 : m - 2);
+}
+
+static SizeType nrStepsPerSweep(SizeType sweep, SizeType m, SizeType mb) {
+  return std::max<SizeType>(0, sweep == m - 2 ? 1 : dlaf::util::ceilDiv(m - sweep - 2, mb));
+}
```

Comment:
@rasolca the first one can be probably moved to our type traits in DLAF, but I'm more interested in the second twos, on how to share them, and if you think it is a good thing, between #413 and this PR.

Moreover, the same discussion can be done for the tests, where similar functions exists (without tile concept).

Since our PRs are strictly related, I think it may be a good idea to share also these functions. Moreover, this relationship may be an idea also for structuring differently this side of the library (instead of backtransformation, it may be worth having tridiagonalization and inside there both the "forward & reverse" operators.

Reply:
After the call:
- I moved `is_complex_v` to `dlaf/types.h`
- I started moving this algo to its own folder `bt_band_to_tri` (name of the actual algo still to be adapted/changed)
- factor out `nrSweeps` and `nrStepsPerSweep` helpers also from the test

About the last point, I realized I was wrong: the same helpers appears both in the algorithm and in the test, there is no difference. I don't know if it is a good idea to share them between test and algo, but for sure we will share them between algorithms #413 and this one.

Looking forward to your comments.